batchflow/models/torch/base.py

```diff
@@ -1555,112 +1542,103 @@ def aggregate_microbatches(self, outputs, chunked_outputs, chunk_sizes, single_o
                 else:
                     output_ = torch.cat([chunk_output[:chunk_size]
                                          for chunk_output, chunk_size in zip(chunked_output, chunk_sizes)], dim=0)
-                result.append(output_)
+                result[output_name] = output_
             else:
                 if isinstance(chunked_output[0], np.ndarray):
-                    result.append(np.mean(chunked_output))
+                    result[output_name] = np.mean(chunked_output)
                 else:
-                    result.append(torch.mean(torch.stack(chunked_output)))
-
-        if single_output:
-            result = result[0]
+                    result[output_name] = torch.mean(torch.stack(chunked_output))
         return result
 
 
-    def compute_outputs(self, predictions):
-        """ Produce additional outputs, defined in the config, from `predictions`.
-        Also adds a number of aliases to predicted tensors.
+    def compute_outputs(self, predictions, operations, targets=None, loss=None):
```

Comment:
I would say that renaming `operations` to `requested_outputs` would make things much easier to understand. Or maybe to `outputs_dict`, to keep in line with the rest of the code